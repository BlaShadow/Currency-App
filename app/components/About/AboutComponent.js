import React, { PureComponent } from 'react'
import { Text, View, Linking } from 'react-native'
import PropTypes from 'prop-types'

import styles from './style'

class LinkComponent extends PureComponent{
    render(){
        return (
            <Text style={{color: 'blue'}}
                  onPress={() => Linking.openURL(this.props.url)}>
                {this.props.children}
            </Text>
        );
    }
}

LinkComponent.propTypes = {
    children: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}

export default class About extends PureComponent {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.title}> Currency Converter </Text>
        <Text style={styles.content}>
            Feel free to use this app, this app it's a test of the React native Tech, 
            if you got any errors send a feedback just contact me at Github 
            <LinkComponent url="https://github.com/BlaShadow"> @blashadow </LinkComponent> or my personal web   
            <LinkComponent url="http://nightdeveloper.net"> Blog </LinkComponent>
        </Text>

        <Text style={styles.content}> All icons and resources are from  
            <LinkComponent url="https://www.flaticon.com/"> Icons </LinkComponent> 
        </Text>

        <Text style={[styles.content, {textAlign: 'center'}]}> 
            Version 1.0
        </Text>
      </View>
    )
  }
}