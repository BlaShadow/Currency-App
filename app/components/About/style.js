import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
    container:{
        height: 240,
        width: '100%',
        padding: 10,
        backgroundColor: 'white',
        borderRadius: 3,
    },
    title:{
        textAlign: 'center',
        width: '100%',
        fontSize: 24,
        fontFamily: 'sans-serif-light'
    },
    content:{
        fontFamily: 'sans-serif-light',
        width: '100%',
        textAlign: 'center',
        fontSize: 16,
        marginTop: 20
    }
});