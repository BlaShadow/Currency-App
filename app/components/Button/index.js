import ClearButton from './ClearButton';
import styles from './style'; 

export {ClearButton, styles};