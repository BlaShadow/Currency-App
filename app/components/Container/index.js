import Container, { Container as ContainerComponent } from './Container';
import styles from './styles';  

export { Container, styles, ContainerComponent };