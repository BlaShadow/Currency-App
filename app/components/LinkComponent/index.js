import { LinkComponent } from './LinkComponent'

export { LinkComponent }