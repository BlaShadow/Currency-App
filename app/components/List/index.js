import ListItem from './ListItem';
import Separator from './separator';

export { ListItem, Separator };