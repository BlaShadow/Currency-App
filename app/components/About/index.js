import AboutComponent from './AboutComponent'
import Styles from './style'

export { AboutComponent, Styles } 