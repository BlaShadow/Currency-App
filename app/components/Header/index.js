import Header from './Header';
import style from './style';

export { Header, style };