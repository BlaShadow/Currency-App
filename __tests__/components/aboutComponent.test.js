import 'react-native';

import React from 'react'
import { AboutComponent } from '../../app/components/About'

import renderer from 'react-test-renderer';

test('Should render About correctly', () => {
    const tree = renderer.create(<AboutComponent />).toJSON();

    expect(tree).toMatchSnapshot();
});
