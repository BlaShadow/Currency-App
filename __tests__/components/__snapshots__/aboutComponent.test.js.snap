// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render About correctly 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
     Currency Converter 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
    Feel free to use this app, this app it's a test of the React native Tech, if you got any errors send a feedback just contact me at Github
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "blue",
        }
      }
    >
       @blashadow 
    </Text>
     or my personal web
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "blue",
        }
      }
    >
       Blog 
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
     All icons and resources are from
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "blue",
        }
      }
    >
       FlatIcon.com 
    </Text>
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        undefined,
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    Version 1.0
  </Text>
</View>
`;
